play/type-racer/app.js

```diff
@@ -261,19 +306,57 @@ function endWordRace() {
     
     // Show stats
     let seconds = 120 - timer;
-    document.getElementById("wpmStat").innerHTML = ((numWords * 60) / seconds).toFixed() + " wpm";
+    let wpm = ((numWords * 60) / seconds);
+    document.getElementById("wpmStat").innerHTML = wpm.toFixed() + " wpm";
     document.getElementById("wordCountStat").innerHTML = numWords;
 
     // set default accuracy to 0
     let wordAccuracyStat = 0.0;
 
     // if the lettersTyped are greater than 0, then calculate a proper wordAccuracyStat
     if (lettersTyped > 0) {
-        wordAccuracyStat = (((lettersTyped - typingErrors) / lettersTyped) * 100).toFixed(1);
+        wordAccuracyStat = (((lettersTyped - typingErrors) / lettersTyped) * 100);
     }
 
     // set the actual wordAccuracyStat in HTML
-    document.getElementById("wordAccuracyStat").innerHTML = wordAccuracyStat + "%";
+    document.getElementById("wordAccuracyStat").innerHTML = wordAccuracyStat.toFixed(1) + "%";
+
+    // calculate current and past playerScore
+    let playerScore = 0;
+    playerScore = (wpm + seconds) * numWords * (wordAccuracyStat / 100);
+    playerScore = playerScore.toFixed();
+    let pastPlayerScore = DM.getItem("score");
+
+    // if getItem() didn't return undefined
+    if (pastPlayerScore != undefined) {
+
+        // convert pastPlayerScore to a valid numeric score
+        pastPlayerScore = JSON.stringify(pastPlayerScore).slice(3);
+        pastPlayerScore = parseInt(pastPlayerScore);
+
+        // save playerScore in DM if new score is higher than the old one
+        if (playerScore > pastPlayerScore) {
+            DM.saveItem("score", playerScore);
+        }
+
+        // display the higher score as the player's high score
+        document.getElementById("playerHighScore").innerHTML = playerScore > pastPlayerScore ? playerScore : pastPlayerScore;
+    }
+
+    // save playerScore, if unable, then popup a warning message
+    else if (DM.saveItem("score", playerScore) == false) {
+        alert("Error: Could not save your score. If you would like to save your score, please login first.");
```

Comment:
I'm not sure if we want to show an alert, just on the off chance it is because the user's browser doesn't support localStorage or something like that.